src/test/java/htsjdk/samtools/cram/structure/CompressionHeaderEncodingMapTest.java

```diff
@@ -53,11 +58,51 @@ public void testAllCRAMEncodingsRoundTripThroughStream() throws IOException {
             final byte[] roundTrippedBytes = baos.toByteArray();
             try (final InputStream is = new ByteArrayInputStream(roundTrippedBytes)) {
                 final CompressionHeaderEncodingMap roundTippedEncodingMap = new CompressionHeaderEncodingMap(is);
-                assertExpectedHTSJDKGeneratedEncodings(roundTippedEncodingMap, Collections.EMPTY_SET);
+                assertExpectedHTSJDKGeneratedEncodings(roundTippedEncodingMap, CompressionHeaderEncodingMap.DATASERIES_NOT_READ_BY_HTSJDK);
             }
         }
     }
 
+    @Test(dataProvider = "readCramTestFiles", description = "make sure that obsolete dataseries, TC and TN are ignored on CRAM read")
+    public void testIgnoreObsoleteDataseriesOnRead(final File sourceFile) throws IOException{
+
+        // iterate through the CRAM file
+        try (InputStream cramInputStream = new BufferedInputStream(new FileInputStream(sourceFile));
+             CountingInputStream countingInputStream = new CountingInputStream(cramInputStream);
+             CramContainerIterator containerIterator = new CramContainerIterator(countingInputStream)) {
+            CramHeader cramHeader = containerIterator.getCramHeader(); //actual cram header
+            final long containerByteOffset = countingInputStream.getCount();
+            do {
```

Comment:
Same comments here as for the other loop (simplify the loop to use the container iterator), and also add finals in the try clause and inside the block.